### DISCUSSION TITLE
is there a script that extract several states in pdb format from Trajectory files?

### DISCUSSION THREAD

**Discussion by Hyunjoo-ibs** [RosettaCommons/RFdiffusion#164] [General]
Hi I am wonder if there a script that can extract several states from Trajectory files.
As the full trajectory file (pdp files in /traj/ folder) has multi-step pdbs in a single trajectory pdb file, I wonder if there is a script that can extract the last (t=1) prediction to t=5 prediction from trajectory pdb file.... 
Or do we have to do that manually in pymol?
Please help me and please let me know if anyone have an idea how to do this..!
Thank you

**Comment by diniramdhni** [comment]
```
import mdtraj as md

# Load trajectory file
traj = md.load('path/to/trajectory.pdb')

# Extract frames from t=1 to t=5
frames_to_extract = list(range(1, 6))  # adjust as needed

# Extract specific frames
selected_frames = traj.slice(frames_to_extract)

# Save the extracted frames to a new trajectory file
selected_frames.save('path/to/extracted_frames.pdb')
```
Before executing this script, make sure to install MDTraj (`pip install mdtraj`).

**Reply by Hyunjoo-ibs** [reply]
Thank you so much!!! I will try and let you know how it goes :) hehe

**Reply by Hyunjoo-ibs** [reply]
Thank you for sharing your script, but for me it kept giving an error like:

Traceback (most recent call last):
  File "test_extract_traj.py", line 4, in <module>
    traj = md.load('/path/to/trajectory.pdb')
  File "/user/.conda/envs/mdtraj_env/lib/python3.7/site-packages/mdtraj/core/trajectory.py", line 433, in load
    value = loader(filename, **kwargs)
  File "/user/.conda/envs/mdtraj_env/lib/python3.7/site-packages/mdtraj/formats/pdb/pdbfile.py", line 153, in load_pdb
    with PDBTrajectoryFile(filename, standard_names=standard_names) as f:
  File "/user/.conda/envs/mdtraj_env/lib/python3.7/site-packages/mdtraj/formats/pdb/pdbfile.py", line 269, in __init__
    self._read_models()
  File "/user/.conda/envs/mdtraj_env/lib/python3.7/site-packages/mdtraj/formats/pdb/pdbfile.py", line 555, in _read_models
    raise ValueError('PDB Error: All MODELs must contain the same number of ATOMs')
ValueError: PDB Error: All MODELs must contain the same number of ATOMs

So I wonder if you also had a experience of getting an error like this... If you had, I would like to ask how you have solved this
Thank you so much for your help!
